rust/src/json_serialize.rs

```diff
@@ -0,0 +1,996 @@
+use std::collections::{BTreeMap, VecDeque};
```

Comment:
We might want to document why we have this instead of using the serde dependency that we use in other parts of the codebase.